src/ts-framework-tests/app/controllers/ShopController.ts

```diff
@@ -1,19 +1,23 @@
 import {Controller, action} from "../../../ts-framework/Controller";
 import {JsonResult} from "../../../ts-framework/View";
+import {HttpController} from "../../../ts-framework/Controller/HttpController";
 
-export class ShopController extends Controller
+export class ShopController extends HttpController
 {
     // GET: /shop
     @action() index()
     {
-        this.response.sendContent("Hello");
+        this.content("Hello");
```

Comment:
You also have to pass the send function pointer to all these structures to keep the controller async.
So the code will be new ContentResult("hello", type, this.send). It's add just too much noise where a single and clean helper function can do it. 

Also on a side note: these are helpers functions, not the only way for the end user to do it. If the end user want to write the more verbose version like you, then he still can do it anyway.

Same comment apply for nexts remarks